### ISSUE TITLE
Failed Installation of ConfigSpace via Pip

### ISSUE THREAD

**Issue by JNC96** [automl/ConfigSpace#130]
OS: Win 10
Python Vers. 3.7.2

Hi! I'm trying to install ConfigSpace but there seems to be an error during installation. Here is the error log:

```
Collecting configspace
  Using cached https://files.pythonhosted.org/packages/be/07/ebb7c6c6d03c82ffe6314161f2c8da32599c841e74b08f5c23812fee0e6f/ConfigSpace-0.4.11.tar.gz
Requirement already satisfied: numpy in c:\windows\system32\venv\lib\site-packages (from configspace) (1.16.4)
Requirement already satisfied: cython in c:\windows\system32\venv\lib\site-packages (from configspace) (0.29.13)
Requirement already satisfied: pyparsing in c:\windows\system32\venv\lib\site-packages (from configspace) (2.4.2)
Building wheels for collected packages: configspace
  Building wheel for configspace (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'c:\windows\system32\venv\scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\jar-b\\AppData\\Local\\Temp\\pip-install-h28j5byo\\configspace\\setup.py'"'"'; __file__='"'"'C:\\Users\\jar-b\\AppData\\Local\\Temp\\pip-install-h28j5byo\\configspace\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\jar-b\AppData\Local\Temp\pip-wheel-f4yhvyxo' --python-tag cp37
       cwd: C:\Users\jar-b\AppData\Local\Temp\pip-install-h28j5byo\configspace\
  Complete output (63 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-3.7
  creating build\lib.win-amd64-3.7\ConfigSpace
  copying ConfigSpace\convert.py -> build\lib.win-amd64-3.7\ConfigSpace
  copying ConfigSpace\exceptions.py -> build\lib.win-amd64-3.7\ConfigSpace
  copying ConfigSpace\__init__.py -> build\lib.win-amd64-3.7\ConfigSpace
  copying ConfigSpace\__version__.py -> build\lib.win-amd64-3.7\ConfigSpace
  creating build\lib.win-amd64-3.7\test
  copying test\test_conditions.py -> build\lib.win-amd64-3.7\test
  copying test\test_configuration_space.py -> build\lib.win-amd64-3.7\test
  copying test\test_forbidden.py -> build\lib.win-amd64-3.7\test
  copying test\test_hyperparameters.py -> build\lib.win-amd64-3.7\test
  copying test\test_util.py -> build\lib.win-amd64-3.7\test
  copying test\__init__.py -> build\lib.win-amd64-3.7\test
  creating build\lib.win-amd64-3.7\ConfigSpace\nx
  copying ConfigSpace\nx\exception.py -> build\lib.win-amd64-3.7\ConfigSpace\nx
  copying ConfigSpace\nx\release.py -> build\lib.win-amd64-3.7\ConfigSpace\nx
  copying ConfigSpace\nx\__init__.py -> build\lib.win-amd64-3.7\ConfigSpace\nx
  creating build\lib.win-amd64-3.7\ConfigSpace\read_and_write
  copying ConfigSpace\read_and_write\irace.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
  copying ConfigSpace\read_and_write\json.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
  copying ConfigSpace\read_and_write\pbs.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
  copying ConfigSpace\read_and_write\pcs.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
  copying ConfigSpace\read_and_write\pcs_new.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
  copying ConfigSpace\read_and_write\pyll.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
  copying ConfigSpace\read_and_write\spearmint_april2013_mod_spearmint_pb2.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
  copying ConfigSpace\read_and_write\__init__.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
  creating build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms
  copying ConfigSpace\nx\algorithms\cycles.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms
  copying ConfigSpace\nx\algorithms\dag.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms
  copying ConfigSpace\nx\algorithms\__init__.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms
  creating build\lib.win-amd64-3.7\ConfigSpace\nx\classes
  copying ConfigSpace\nx\classes\digraph.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\classes
  copying ConfigSpace\nx\classes\graph.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\classes
  copying ConfigSpace\nx\classes\__init__.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\classes
  creating build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms\components
  copying ConfigSpace\nx\algorithms\components\strongly_connected.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms\components
  copying ConfigSpace\nx\algorithms\components\__init__.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms\components
  creating build\lib.win-amd64-3.7\test\read_and_write
  copying test\read_and_write\test_irace_writer.py -> build\lib.win-amd64-3.7\test\read_and_write
  copying test\read_and_write\test_json.py -> build\lib.win-amd64-3.7\test\read_and_write
  copying test\read_and_write\test_pb_converter.py -> build\lib.win-amd64-3.7\test\read_and_write
  copying test\read_and_write\test_pcs_converter.py -> build\lib.win-amd64-3.7\test\read_and_write
  copying test\read_and_write\test_pyll_util.py -> build\lib.win-amd64-3.7\test\read_and_write
  copying test\read_and_write\__init__.py -> build\lib.win-amd64-3.7\test\read_and_write
  creating build\lib.win-amd64-3.7\test\test_converters_and_test_searchspaces
  copying test\test_converters_and_test_searchspaces\test_sample_configuration_spaces.py -> build\lib.win-amd64-3.7\test\test_converters_and_test_searchspaces
  copying test\test_converters_and_test_searchspaces\__init__.py -> build\lib.win-amd64-3.7\test\test_converters_and_test_searchspaces
  creating build\lib.win-amd64-3.7\test\test_searchspaces
  copying test\test_searchspaces\__init__.py -> build\lib.win-amd64-3.7\test\test_searchspaces
  running build_ext
  skipping 'ConfigSpace\hyperparameters.c' Cython extension (up-to-date)
  building 'ConfigSpace.hyperparameters' extension
  creating build\temp.win-amd64-3.7
  creating build\temp.win-amd64-3.7\Release
  creating build\temp.win-amd64-3.7\Release\ConfigSpace
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IC:\Users\jar-b\AppData\Local\Programs\Python\Python37\include -IC:\Users\jar-b\AppData\Local\Programs\Python\Python37\include -Ic:\windows\system32\venv\lib\site-packages\numpy\core\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt" /TcConfigSpace\hyperparameters.c /Fobuild\temp.win-amd64-3.7\Release\ConfigSpace\hyperparameters.obj
  hyperparameters.c
  ConfigSpace\hyperparameters.c(598): fatal error C1083: Cannot open include file: 'numpy/arrayobject.h': No such file or directory
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
  ----------------------------------------
  ERROR: Failed building wheel for configspace
  Running setup.py clean for configspace
Failed to build configspace
Installing collected packages: configspace
  Running setup.py install for configspace ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\windows\system32\venv\scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\jar-b\\AppData\\Local\\Temp\\pip-install-h28j5byo\\configspace\\setup.py'"'"'; __file__='"'"'C:\\Users\\jar-b\\AppData\\Local\\Temp\\pip-install-h28j5byo\\configspace\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\jar-b\AppData\Local\Temp\pip-record-g292rq_s\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\windows\system32\venv\include\site\python3.7\configspace'
         cwd: C:\Users\jar-b\AppData\Local\Temp\pip-install-h28j5byo\configspace\
    Complete output (63 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.7
    creating build\lib.win-amd64-3.7\ConfigSpace
    copying ConfigSpace\convert.py -> build\lib.win-amd64-3.7\ConfigSpace
    copying ConfigSpace\exceptions.py -> build\lib.win-amd64-3.7\ConfigSpace
    copying ConfigSpace\__init__.py -> build\lib.win-amd64-3.7\ConfigSpace
    copying ConfigSpace\__version__.py -> build\lib.win-amd64-3.7\ConfigSpace
    creating build\lib.win-amd64-3.7\test
    copying test\test_conditions.py -> build\lib.win-amd64-3.7\test
    copying test\test_configuration_space.py -> build\lib.win-amd64-3.7\test
    copying test\test_forbidden.py -> build\lib.win-amd64-3.7\test
    copying test\test_hyperparameters.py -> build\lib.win-amd64-3.7\test
    copying test\test_util.py -> build\lib.win-amd64-3.7\test
    copying test\__init__.py -> build\lib.win-amd64-3.7\test
    creating build\lib.win-amd64-3.7\ConfigSpace\nx
    copying ConfigSpace\nx\exception.py -> build\lib.win-amd64-3.7\ConfigSpace\nx
    copying ConfigSpace\nx\release.py -> build\lib.win-amd64-3.7\ConfigSpace\nx
    copying ConfigSpace\nx\__init__.py -> build\lib.win-amd64-3.7\ConfigSpace\nx
    creating build\lib.win-amd64-3.7\ConfigSpace\read_and_write
    copying ConfigSpace\read_and_write\irace.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
    copying ConfigSpace\read_and_write\json.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
    copying ConfigSpace\read_and_write\pbs.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
    copying ConfigSpace\read_and_write\pcs.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
    copying ConfigSpace\read_and_write\pcs_new.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
    copying ConfigSpace\read_and_write\pyll.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
    copying ConfigSpace\read_and_write\spearmint_april2013_mod_spearmint_pb2.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
    copying ConfigSpace\read_and_write\__init__.py -> build\lib.win-amd64-3.7\ConfigSpace\read_and_write
    creating build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms
    copying ConfigSpace\nx\algorithms\cycles.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms
    copying ConfigSpace\nx\algorithms\dag.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms
    copying ConfigSpace\nx\algorithms\__init__.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms
    creating build\lib.win-amd64-3.7\ConfigSpace\nx\classes
    copying ConfigSpace\nx\classes\digraph.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\classes
    copying ConfigSpace\nx\classes\graph.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\classes
    copying ConfigSpace\nx\classes\__init__.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\classes
    creating build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms\components
    copying ConfigSpace\nx\algorithms\components\strongly_connected.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms\components
    copying ConfigSpace\nx\algorithms\components\__init__.py -> build\lib.win-amd64-3.7\ConfigSpace\nx\algorithms\components
    creating build\lib.win-amd64-3.7\test\read_and_write
    copying test\read_and_write\test_irace_writer.py -> build\lib.win-amd64-3.7\test\read_and_write
    copying test\read_and_write\test_json.py -> build\lib.win-amd64-3.7\test\read_and_write
    copying test\read_and_write\test_pb_converter.py -> build\lib.win-amd64-3.7\test\read_and_write
    copying test\read_and_write\test_pcs_converter.py -> build\lib.win-amd64-3.7\test\read_and_write
    copying test\read_and_write\test_pyll_util.py -> build\lib.win-amd64-3.7\test\read_and_write
    copying test\read_and_write\__init__.py -> build\lib.win-amd64-3.7\test\read_and_write
    creating build\lib.win-amd64-3.7\test\test_converters_and_test_searchspaces
    copying test\test_converters_and_test_searchspaces\test_sample_configuration_spaces.py -> build\lib.win-amd64-3.7\test\test_converters_and_test_searchspaces
    copying test\test_converters_and_test_searchspaces\__init__.py -> build\lib.win-amd64-3.7\test\test_converters_and_test_searchspaces
    creating build\lib.win-amd64-3.7\test\test_searchspaces
    copying test\test_searchspaces\__init__.py -> build\lib.win-amd64-3.7\test\test_searchspaces
    running build_ext
    skipping 'ConfigSpace\hyperparameters.c' Cython extension (up-to-date)
    building 'ConfigSpace.hyperparameters' extension
    creating build\temp.win-amd64-3.7
    creating build\temp.win-amd64-3.7\Release
    creating build\temp.win-amd64-3.7\Release\ConfigSpace
    C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IC:\Users\jar-b\AppData\Local\Programs\Python\Python37\include -IC:\Users\jar-b\AppData\Local\Programs\Python\Python37\include -Ic:\windows\system32\venv\lib\site-packages\numpy\core\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt" /TcConfigSpace\hyperparameters.c /Fobuild\temp.win-amd64-3.7\Release\ConfigSpace\hyperparameters.obj
    hyperparameters.c
    ConfigSpace\hyperparameters.c(598): fatal error C1083: Cannot open include file: 'numpy/arrayobject.h': No such file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\windows\system32\venv\scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\jar-b\\AppData\\Local\\Temp\\pip-install-h28j5byo\\configspace\\setup.py'"'"'; __file__='"'"'C:\\Users\\jar-b\\AppData\\Local\\Temp\\pip-install-h28j5byo\\configspace\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\jar-b\AppData\Local\Temp\pip-record-g292rq_s\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\windows\system32\venv\include\site\python3.7\configspace' Check the logs for full command output.
```

I've already install NumPy (1.16.4), Cython (0.29.13) and PyParsing (2.4.2). I've also got Visual Studio 2017 installed, and Typing for good measure. I'm not really sure what the error is, but the first trigger seems to the the hyperparameters.c file, and there's another error under that pointing to Microsoft Visual Studio. I'm not sure how to fix the issue, though.

TIA!

**Comment by PhilipMay** [comment]
I have a similar problem. I am on linux, try to install hpbandster. This pip package is a dependency but I get an error message with pip:

```
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/ConfigSpace
  copying ConfigSpace/exceptions.py -> build/lib.linux-x86_64-3.6/ConfigSpace
  copying ConfigSpace/convert.py -> build/lib.linux-x86_64-3.6/ConfigSpace
  copying ConfigSpace/__init__.py -> build/lib.linux-x86_64-3.6/ConfigSpace
  copying ConfigSpace/__version__.py -> build/lib.linux-x86_64-3.6/ConfigSpace
  creating build/lib.linux-x86_64-3.6/test
  copying test/test_hyperparameters.py -> build/lib.linux-x86_64-3.6/test
  copying test/test_conditions.py -> build/lib.linux-x86_64-3.6/test
  copying test/test_configuration_space.py -> build/lib.linux-x86_64-3.6/test
  copying test/test_forbidden.py -> build/lib.linux-x86_64-3.6/test
  copying test/test_util.py -> build/lib.linux-x86_64-3.6/test
  copying test/__init__.py -> build/lib.linux-x86_64-3.6/test
  creating build/lib.linux-x86_64-3.6/ConfigSpace/read_and_write
  copying ConfigSpace/read_and_write/pcs_new.py -> build/lib.linux-x86_64-3.6/ConfigSpace/read_and_write
  copying ConfigSpace/read_and_write/irace.py -> build/lib.linux-x86_64-3.6/ConfigSpace/read_and_write
  copying ConfigSpace/read_and_write/spearmint_april2013_mod_spearmint_pb2.py -> build/lib.linux-x86_64-3.6/ConfigSpace/read_and_write
  copying ConfigSpace/read_and_write/json.py -> build/lib.linux-x86_64-3.6/ConfigSpace/read_and_write
  copying ConfigSpace/read_and_write/pbs.py -> build/lib.linux-x86_64-3.6/ConfigSpace/read_and_write
  copying ConfigSpace/read_and_write/pcs.py -> build/lib.linux-x86_64-3.6/ConfigSpace/read_and_write
  copying ConfigSpace/read_and_write/__init__.py -> build/lib.linux-x86_64-3.6/ConfigSpace/read_and_write
  copying ConfigSpace/read_and_write/pyll.py -> build/lib.linux-x86_64-3.6/ConfigSpace/read_and_write
  creating build/lib.linux-x86_64-3.6/ConfigSpace/nx
  copying ConfigSpace/nx/exception.py -> build/lib.linux-x86_64-3.6/ConfigSpace/nx
  copying ConfigSpace/nx/release.py -> build/lib.linux-x86_64-3.6/ConfigSpace/nx
  copying ConfigSpace/nx/__init__.py -> build/lib.linux-x86_64-3.6/ConfigSpace/nx
  creating build/lib.linux-x86_64-3.6/ConfigSpace/nx/classes
  copying ConfigSpace/nx/classes/graph.py -> build/lib.linux-x86_64-3.6/ConfigSpace/nx/classes
  copying ConfigSpace/nx/classes/__init__.py -> build/lib.linux-x86_64-3.6/ConfigSpace/nx/classes
  copying ConfigSpace/nx/classes/digraph.py -> build/lib.linux-x86_64-3.6/ConfigSpace/nx/classes
  creating build/lib.linux-x86_64-3.6/ConfigSpace/nx/algorithms
  copying ConfigSpace/nx/algorithms/cycles.py -> build/lib.linux-x86_64-3.6/ConfigSpace/nx/algorithms
  copying ConfigSpace/nx/algorithms/dag.py -> build/lib.linux-x86_64-3.6/ConfigSpace/nx/algorithms
  copying ConfigSpace/nx/algorithms/__init__.py -> build/lib.linux-x86_64-3.6/ConfigSpace/nx/algorithms
  creating build/lib.linux-x86_64-3.6/ConfigSpace/nx/algorithms/components
  copying ConfigSpace/nx/algorithms/components/strongly_connected.py -> build/lib.linux-x86_64-3.6/ConfigSpace/nx/algorithms/components
  copying ConfigSpace/nx/algorithms/components/__init__.py -> build/lib.linux-x86_64-3.6/ConfigSpace/nx/algorithms/components
  creating build/lib.linux-x86_64-3.6/test/test_converters_and_test_searchspaces
  copying test/test_converters_and_test_searchspaces/test_sample_configuration_spaces.py -> build/lib.linux-x86_64-3.6/test/test_converters_and_test_searchspaces
  copying test/test_converters_and_test_searchspaces/__init__.py -> build/lib.linux-x86_64-3.6/test/test_converters_and_test_searchspaces
  creating build/lib.linux-x86_64-3.6/test/test_searchspaces
  copying test/test_searchspaces/__init__.py -> build/lib.linux-x86_64-3.6/test/test_searchspaces
  creating build/lib.linux-x86_64-3.6/test/read_and_write
  copying test/read_and_write/test_pb_converter.py -> build/lib.linux-x86_64-3.6/test/read_and_write
  copying test/read_and_write/test_json.py -> build/lib.linux-x86_64-3.6/test/read_and_write
  copying test/read_and_write/test_pyll_util.py -> build/lib.linux-x86_64-3.6/test/read_and_write
  copying test/read_and_write/test_pcs_converter.py -> build/lib.linux-x86_64-3.6/test/read_and_write
  copying test/read_and_write/__init__.py -> build/lib.linux-x86_64-3.6/test/read_and_write
  copying test/read_and_write/test_irace_writer.py -> build/lib.linux-x86_64-3.6/test/read_and_write
  running build_ext
  building 'ConfigSpace.hyperparameters' extension
  error: unknown file type '.pyx' (from 'ConfigSpace/hyperparameters.pyx')
  ----------------------------------------
  ERROR: Failed building wheel for ConfigSpace
```

**Comment by jwmueller** [comment]
Same issue here

**Comment by mfeurer** [comment]
That's definitively the solution for the issue observed by @PhilipMay (thanks @jwmueller ). Does it also solve the installation problem on Windows? (I can't help there because I'm not a windows user).

**Comment by JNC96** [comment]
I've installed Cython beforehand and pip seems to detect it.

```
Requirement already satisfied: cython in c:\windows\system32\venv\lib\site-packages (from configspace) (0.29.13)
```

Haven't had much luck since posting this either.

**Comment by anhdungle93** [comment]
I faced the same problem as well when i tried to install hpbandster, I would love to know how to solve it

**Comment by bombhero** [comment]
OS: Win10
Python: 3.7.3
I meet the same issue. And I have installed cython.

**Comment by ilyaivensky** [comment]
Same issue
OSx 10.13.6
Python: 3.7.5

**Comment by wonka929** [comment]
Same here! 
OS: Win10
Python: 3.7.5

**Comment by tadhgfitzgerald** [comment]
A bit late to the party but I was encountering the same issue on OSx 10.15.3 and python 3.7. Downloading the latest release from GitHub and building/installing by setup.py worked, not a fix but a workaround for those stuck. Hope that helps anyone who ends up here.

**Comment by DejanKostyszyn** [comment]
`sudo apt install python-dev` should do it for debian based systems.

**Comment by johncmerfeld** [comment]
@jwmueller's solution worked for me

**Comment by saiprasadbarke** [comment]
> A bit late to the party but I was encountering the same issue on OSx 10.15.3 and python 3.7. Downloading the latest release from GitHub and building/installing by setup.py worked, not a fix but a workaround for those stuck. Hope that helps anyone who ends up here.

This worked for me. Thankyou.

**Comment by shahinshirazi1** [comment]
I am having the same issue:
OS: Win10
Python: tried 3.7 and 3.5. 
I am getting the following error: error:
 Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": https://visualstudio.microsoft.com/downloads/ 
I installed Microsoft Visual Studio and still getting the same error.

**Comment by labenech** [comment]
pip install Cython before pip install other requirements solve the problem for me !

**Comment by microprediction** [comment]
Same problem. 
Python 3.8.6
Big Sur 11.2

Also fails with:
pip install git+https://github.com/auto-flow/ultraopt.git

**Comment by ScDor** [comment]
Same with WSL2 on windows 10

**Comment by Previatto** [comment]
I had this problem on Ubuntu 16.04, python 3.7.1

Couldn't install ConfigSpace 0.4.18, after uninstalling "typing" package pip install ConfigSpace==0.4.18 worked

**Comment by quuhua911** [comment]
nobody answers or settles the problems？

**Comment by eddiebergman** [comment]
Hi all,

Hopefully this issue will be solved by #224 which adds wheels for mac/windows and linux which has everything precompiled. I will comment back here once it is released on Pypi and installable with `pip`.

Best,
Eddie
